File: sadues-server/src/main/java/com/elroykanye/sadues/business/service/i/UniversityService.java
package com.elroykanye.sadues.business.service.i;

import com.elroykanye.sadues.api.dto.UniversityDto;
import com.elroykanye.sadues.api.dto.response.SaResponse;
import com.elroykanye.sadues.data.entity.University;

import java.util.List;

public interface UniversityService {
    SaResponse create( UniversityDto dto);
    University getEntity(long id);
    UniversityDto getDto(long id);
    List<University> getAllEntities();
    List<UniversityDto> getAllDto();
    SaResponse update( UniversityDto dto);
    void delete(long id);
}

File: src/main/java/com/elroykanye/sadues/business/service/i/UniversityService.kt
package com.elroykanye.sadues.business.service.i

import com.elroykanye.sadues.api.dto.UniversityDto
import com.elroykanye.sadues.api.dto.response.SaResponse
import com.elroykanye.sadues.data.entity.University

interface UniversityService {
    fun create(universityDto: UniversityDto): SaResponse

    fun getEntity(id: Long): University

    fun getDto(id: Long): UniversityDto

    fun getAllEntities(): List<University>

    fun getAllDto(): List<UniversityDto>

    fun update(universityDto: UniversityDto): SaResponse

    fun delete(id: Long)
}
